ring/basic_lifecycler.go

```diff
@@ -316,7 +316,8 @@ func (l *BasicLifecycler) registerInstance(ctx context.Context) error {
 		// Always overwrite the instance in the ring (even if already exists) because some properties
 		// may have changed (stated, tokens, zone, address) and even if they didn't the heartbeat at
 		// least did.
-		instanceDesc = ringDesc.AddIngester(l.cfg.ID, l.cfg.Addr, l.cfg.Zone, tokens, state, registeredAt)
+		// TODO: add support for read-only state and timestamp.
```

Comment:
Do you plan to address these TODOs soon? If no, I would rather update the `BasicLifecycler struct` go doc to mention that this lifecycler doesn't support read-only state for now.

Reply:
Converted these to a doc comment as suggested.